src/algebra/big_operators/finprod.lean

```diff
@@ -0,0 +1,631 @@
+There is work on `fincard` in the pipeline, which returns the cardinality of `X` if it
+is finite, and 0 otherwise.
+
+## Tags
+
+finsum, finprod, finite sum, finite product
+-/
+
+open_locale classical
+open function set
+
+/-!
+### Definition and relation to `finset.sum` and `finset.prod`
+-/
+
+section sort
+variables {α β ι : Sort*} {M N : Type*} [comm_monoid M] [comm_monoid N]
+
+open_locale big_operators
+
+/-- Sum of `f x` as `x` ranges over the elements of the support of `f`, if it's finite. Zero
+otherwise. -/
+@[irreducible] noncomputable def finsum {M} [add_comm_monoid M] (f : α → M) : M :=
+if h : finite (support (f ∘ plift.down)) then ∑ i in h.to_finset, f i.down else 0
+
+/-- Product of `f x` as `x` ranges over the elements of the multiplicative support of `f`, if it's
+finite. One otherwise. -/
+@[irreducible, to_additive]
+noncomputable def finprod (f : α → M) : M :=
+if h : finite (mul_support (f ∘ plift.down)) then ∏ i in h.to_finset, f i.down else 1
+
+localized "notation `∑ᶠ` binders `, ` r:(scoped:67 f, finsum f) := r" in big_operators
+
+localized "notation `∏ᶠ` binders `, ` r:(scoped:67 f, finprod f) := r" in big_operators
+
+@[to_additive] lemma finprod_eq_prod_plift_of_mul_support_to_finset_subset
+  {f : α → M} (hf : finite (mul_support (f ∘ plift.down))) {s : finset (plift α)}
+  (hs : hf.to_finset ⊆ s) :
+  ∏ᶠ i, f i = ∏ i in s, f i.down :=
+begin
+  rw [finprod, dif_pos hf],
+  refine finset.prod_subset hs (λ x hx hxf, _),
+  rwa [hf.mem_to_finset, nmem_mul_support] at hxf
+end
+
+@[to_additive] lemma finprod_eq_prod_plift_of_mul_support_subset
+  {f : α → M} {s : finset (plift α)} (hs : mul_support (f ∘ plift.down) ⊆ s) :
+  ∏ᶠ i, f i = ∏ i in s, f i.down :=
+finprod_eq_prod_plift_of_mul_support_to_finset_subset
+  (s.finite_to_set.subset hs) $ λ x hx, by { rw finite.mem_to_finset at hx, exact hs hx }
+
+@[simp, to_additive] lemma finprod_one : ∏ᶠ i : α, (1 : M) = 1 :=
+begin
+  have : mul_support (λ x : plift α, (λ _, 1 : α → M) x.down) ⊆ (∅ : finset (plift α)),
+    from λ x h, h rfl,
+  rw [finprod_eq_prod_plift_of_mul_support_subset this, finset.prod_empty]
+end
+
+@[to_additive] lemma finprod_of_empty (ha : α → false) (f : α → M) : ∏ᶠ i, f i = 1 :=
+by { rw ← finprod_one, congr' with x, exact (ha x).elim }
+
+@[simp, to_additive] lemma finprod_false (f : false → M) : ∏ᶠ i, f i = 1 :=
+finprod_of_empty id _
+
+@[to_additive] lemma finprod_unique [unique α] (f : α → M) : ∏ᶠ i, f i = f (default α) :=
+begin
+  have : mul_support (f ∘ plift.down) ⊆ (finset.univ : finset (plift α)),
+    from λ x _, finset.mem_univ x,
+  rw [finprod_eq_prod_plift_of_mul_support_subset this, univ_unique,
+    finset.prod_singleton],
+  exact congr_arg f (plift.down_up _)
+end
+
+@[simp, to_additive] lemma finprod_true (f : true → M) : ∏ᶠ i, f i = f trivial :=
+@finprod_unique true M _ ⟨⟨trivial⟩, λ _, rfl⟩ f
+
+@[to_additive] lemma finprod_eq_dif {p : Prop} (f : p → M) :
```

Comment:
This is missing a `decidable p` argument

Reply:
I've gone ahead and pushed a commit that fixes this throughout.

Reply:
Thanks!